courtjester: support playing from spotify links

anayw2001 opened the pull request.

anayw2001 committed:
courtjester: support playing from spotify links

bdashore3 commented:
Please allow maintainers to edit commits on this PR. There are some things that need fixing.

anayw2001 committed:
courtjester: pull credentials from info.json

anayw2001 committed:
CourtJester: add artist to spotify search query

kingbri1 committed:
Music: Simplify aspects of spotify fetching

1. An OAuth token is not needed to fetch a track from Spotify. Remove
this process and just fetch via the client.

2. Return an Option with the track name instead of an empty string. This
prevents an empty string from being tossed around and we can match
on the option to bail if the track isn't found.

Signed-off-by: kingbri <bdashore3@gmail.com>

bdashore3 commented:
One more commit needs to be added to fix the URL splitting then a merge should be good.

kingbri1 reviewed (approved):
Looks good now. Will merge once checks finish.
